snudown.h

```diff
@@ -0,0 +1,118 @@
+/* snudown.h - header for snudown */
```

Comment:
While this works for the python module form, it's not well structured and will cause maintenance problems in a few years.

Please put all of the implementations in a separate file. Maybe `renderers.c`?

The new files will need to be mentioned in `fuzzing/CMakeLists.txt` to  allow snudown-validator to be fuzz tested.